src/backend/core/management/commands/reindex_with_embedding.py

```diff
@@ -0,0 +1,171 @@
+"""
+Handle reindexing of documents with embeddings in OpenSearch.
+"""
+
+from django.core.management.base import BaseCommand, CommandError
+
+from opensearchpy.exceptions import NotFoundError
+
+from core.services.opensearch import (
+    check_hybrid_search_enabled,
+    embed_text,
+    ensure_index_exists,
+    opensearch_client,
+)
+
+
+class Command(BaseCommand):
+    """Reindex all documents with embeddings"""
+
+    help = __doc__
+    opensearch_client_ = opensearch_client()
+
+    def add_arguments(self, parser):
+        parser.add_argument("index_name", type=str)
+
+    def handle(self, *args, **options):
+        """Launch the reindexing with embedding."""
+
+        source_index = options["index_name"]
+        # destination index is used to duplicate the source index with embeddings.
+        # It is then rename to the source index name after reindexing.
+        destination_index = f"{source_index}-embedded"
+
+        if not check_hybrid_search_enabled():
+            raise CommandError("Hybrid search is not enabled or properly configured.")
+
+        try:
+            self.opensearch_client_.indices.get(index=source_index)
+        except NotFoundError as error:
+            raise CommandError(
+                f"Source index {source_index} does not exist."
+            ) from error
+
+        self.stdout.write(f"[INFO] Start reindexing {source_index} with embedding.")
+
+        reindex_with_embedding(self.opensearch_client_, source_index, destination_index)
+
+        self.opensearch_client_.indices.refresh(index=destination_index)
+        # Validate that reindexing completed successfully
+        # before renaming destination index to source index.
+        self.validate_reindex(source_index, destination_index)
+        self.replace_source_index_with_destination_index(
+            source_index, destination_index
+        )
+
+        self.stdout.write(f"[INFO] Reindexing of {source_index} is successful.")
+
+    def validate_reindex(self, source_index, destination_index):
```

Comment:
You may not need that, if you use the way I suggest before:
 - if you run the command once, the documents are updated
 - if you run the command again, no document should be updated
 - (and if the command is tested it should have done its duty ^^)